### PR TITLE
Try to avoid to interpret wheel events as pinch-to-zoom ones (bug 1810800)

### PR DESCRIPTION
**Pull request by calixteman**
The solution here is likely far from perfect but as far as I can tell it works "correctly" on linux/mac/windows.
With Firefox, we've potentially a bug on laptops when the pref `mousewheel.with_control.action` is set to 1 (which means that we want to scroll even if ctrl is pressed). So we should hack on Firefox to find a satisfying solution but as said it's probably the best we can do.

### CODE REVIEW

**Review by Snuffleupagus — APPROVED** [review]
r=me, since this seems to work in my testing when using mouse-wheel zooming on Windows.

However, it really doesn't feel great having to jump through these kind of hoops and there's probably going to be *some* configuration where it still doesn't work correctly.

I wonder what our chances are of getting e.g. Firefox to add a property to the `WheelEvent` that would let us detect this particular case directly?

File: web/app.js [hunk 1]
```diff
@@ -2659,7 +2659,26 @@ function webViewerWheel(evt) {
   // https://developer.mozilla.org/en-US/docs/Web/API/WheelEvent#browser_compatibility
   // Hence if ctrlKey is true but ctrl key hasn't been pressed then we can
   // infer that we have a pinch-to-zoom.
-  const isPinchToZoom = evt.ctrlKey && !PDFViewerApplication._isCtrlKeyDown;
+  // But the ctrlKey could have been pressed outside of the browser windows,
+  // hence we try to do some magic to guess if the scaleFactor is likely coming
+  // from a pinch-to zoom or not.
```
Comment (Snuffleupagus) [inline web/app.js:2664]: Nits:
```suggestion
  // But the ctrlKey could have been pressed outside of the browser window,
  // hence we try to do some magic to guess if the scaleFactor is likely coming
  // from a pinch-to-zoom or not.
```

**Comment by calixteman** [comment]
> r=me, since this seems to work in my testing when using mouse-wheel zooming on Windows.
> 
> However, it really doesn't feel great having to jump through these kind of hoops and there's probably going to be _some_ configuration where it still doesn't work correctly.
> 

Yep it's why it's far from perfect.
We can try to add some workarounds based on mouse-enter/leave.

> I wonder what our chances are of getting e.g. Firefox to add a property to the `WheelEvent` that would let us detect this particular case directly?

Yes it's exactly what I've in mind :).

**Pull request merged**